src/content/blog/podman-and-docker-rootless.md

```diff
@@ -0,0 +1,481 @@
+
+- [Podman run/build performance issues](https://github.com/containers/podman/issues/13226)
+- [Podman Performance documentation](https://github.com/containers/podman/blob/main/docs/tutorials/performance.md)
+
+To improve performance, install `fuse-overlayfs` and configure the overlay storage driver:
+
+**Install fuse-overlayfs**:
+
+```bash
+# Ubuntu/Debian
+sudo apt-get install fuse-overlayfs
+# Fedora
+sudo dnf install fuse-overlayfs
+# Arch Linux
+sudo pacman -S fuse-overlayfs
+```
+
+**Configure storage**:
+
+```bash
+cat << 'EOF' > ~/.config/containers/storage.conf
+[storage]
+driver = "overlay"
+[storage.options.overlay]
+mount_program = "/usr/bin/fuse-overlayfs"
+EOF
+```
+
+**Warning**: If you already have Podman containers, images, or volumes, you'll need to reset Podman for this change to take effect:
+
+```bash
+podman system reset
+```
+
+This removes all existing containers, images, and volumes (similar to `docker system prune -a`).
+
+## Setting Up Docker Rootless with DDEV
+
+Docker Rootless offers rootless security with full Docker compatibility.
+
+### Installation
+
+1. Follow the official [Docker Rootless installation guide](https://docs.docker.com/engine/security/rootless/).
+
+2. For daemon configuration, see the [Docker daemon documentation](https://docs.docker.com/engine/daemon/).
+
+3. Configure system for privileged port access:
+
+   ```bash
+   # Allow binding to ports below 1024
+   echo 'net.ipv4.ip_unprivileged_port_start=0' | sudo tee -a /etc/sysctl.d/60-rootless.conf
+
+   # Apply without rebooting
+   sudo sysctl -p /etc/sysctl.d/60-rootless.conf
+   ```
+
+4. Enable and start the Docker socket:
+
+   ```bash
+   systemctl --user enable --now docker.socket
+   ```
+
+5. Verify the socket:
+
+   ```bash
+   ls $XDG_RUNTIME_DIR/docker.sock
+   ```
+
+   You should see `/run/user/1000/docker.sock` (the number may vary).
+
+6. Configure Docker context (if needed):
+
+   ```bash
+   # View existing contexts
+   docker context ls
+
+   # Create rootless context
+   docker context create rootless \
+       --description "Rootless runtime socket" \
+       --docker host="unix://$XDG_RUNTIME_DIR/docker.sock"
+
+   # Switch to the context
+   docker context use rootless
+
+   # Verify it works
+   docker ps
+   ```
+
+7. Docker Rootless requires no-bind-mounts mode
+
+   Docker Rootless has a limitation with bind mounts that affects DDEV. You must enable `no-bind-mounts` mode:
+
+   ```bash
+   ddev config global --no-bind-mounts=true
+   ```
+
+   **Why this is needed**:
+
+   Docker Rootless sets ownership for bind mounts to `root` inside containers. This is a known issue:
+   - [Mounting a volume with rootless always assigns ownership to root](https://github.com/moby/moby/issues/45919)
+   - [Add ability to mount volume as user other than root](https://github.com/moby/moby/issues/2259)
+
+   The `root` user inside the container maps to your host user, but many services will not run as root:
```

Comment:
It's interesting that this is the same classic problem we've had with Docker Desktop for Linux and for a time with virtiofs.